Assets/Code/Interactables/Effects/Static Effects/Modifiers/AttachedModifier.cs

```diff
@@ -0,0 +1,22 @@
+﻿public abstract class AttachedModifier : AttachedStaticEffect {
+    public ModifierType modifierType;
+
+    // How much modifier is currently applied
+    public int amount;
+
+    // How much to change the modifier per turn (can be positive or negative)
+    // Can be used to make "temporary buffs" that last until EOT or statuses like DOT that decrement slightly over time
+    public int incrementAmount;
+
+    // This boolean controls whether a modifier should be removed if the amount ever falls below zero
+    // This can be useful for temporary modifiers that decrement by an amount every turn and should be removed after the amount reaches zero
+    // Otherwise, amount would become negative and the character would get persisting negative effects
+    // Obviously sometimes it would be useful to have a decrementing amount persist, but this is the control field for that
+    public bool removeOnZero;
+    public AttachedModifier(Fighter character, int id, ModifierType modifierType, int amount, int incrementAmount, bool removeOnZero) : base(character, id, modifierType.ToString()) {
+        this.modifierType = modifierType;
+        this.amount = amount;
+        this.incrementAmount = incrementAmount;
+        this.removeOnZero = removeOnZero;
```

Comment:
not a fan of this implementation, too specific. doesn't allow for:
- increase by 1 a turn until it's +3, then stop (and stay permanently at +3)
- decrease by 1 a turn until it's -10, then go away (reset to zero)
- increase by x2 every turn
- increase by 1 every other turn
would recommend just taking out incrementAmount and removeOnZero for now, then implementing those via means of another static method that functions as an 'effectCount'

Reply:
yeah it's probably better just to separate the logic into separate classes.